cpp/src/arrow/dataset/dataset_writer.cc

```diff
@@ -174,19 +174,9 @@ class DatasetWriterFileQueue {
   }
 
   void ScheduleBatch(std::shared_ptr<RecordBatch> batch) {
-    // TODO(ARROW-17110) This task is only here to std::move the batch
-    // It can be a lambda once we adopt C++14
-    struct WriteTask : public util::AsyncTaskScheduler::Task {
-      WriteTask(DatasetWriterFileQueue* self, std::shared_ptr<RecordBatch> batch)
-          : self(self), batch(std::move(batch)) {}
-      Result<Future<>> operator()(util::AsyncTaskScheduler*) {
-        return self->WriteNext(std::move(batch));
-      }
-      DatasetWriterFileQueue* self;
-      std::shared_ptr<RecordBatch> batch;
-    };
-    scheduler_->AddTask(
-        ::arrow::internal::make_unique<WriteTask>(this, std::move(batch)));
+    scheduler_->AddSimpleTask([self = this, batch = std::move(batch)]() {
```

Comment:
Do we want to specify all arguments in this way or just the ones that we need to capture specially?  E.g.  Is `[self = this, batch = std::move(batch)]` preferred over `[this, batch = std::move(batch)]`?

Reply:
That's a good question. I think either way is ok.